src/document/Document.js

```diff
@@ -758,6 +764,27 @@ define(function (require, exports, module) {
         return this.file instanceof InMemoryFile;
     };
 
+    /**
+     * Set lineEnding of the document. Change Text in the process.
+     */
+    Document.prototype.setLineEndings = function (lineEndings) {
+        if (this._lineEndings === lineEndings) {
+            return;
+        }
+        this._lineEndings = lineEndings;
```

Comment:
If this is toggled twice, the document is effectively in unedited state and hence should not throw a save/don't save prompt while closing the file.

Other thing is should the change of line ending be part of undo stack?